src/sentry/issues/producer.py

```diff
@@ -64,7 +65,14 @@ def produce_occurrence_to_kafka(
     if payload_data is None:
         return
 
-    payload = KafkaPayload(None, json.dumps(payload_data).encode("utf-8"), [])
+    partition_key = None
+    if (
+        options.get("issue_platform.use_kafka_partition_key")
+        and occurrence
+        and occurrence.fingerprint
+    ):
+        partition_key = bytes(occurrence.fingerprint[0], "utf-8")
+    payload = KafkaPayload(partition_key, json.dumps(payload_data).encode("utf-8"), [])
```

Comment:
are we worried about a single fingerprint overwhelming one of the consumers? could that be possible?

Reply:
@JoshFerge I don't know, let me check with the experts

Reply:
It's a possibility but I don't think it will be a problem in practice